core/domain/topic_domain.py

```diff
@@ -347,22 +500,158 @@ def update_additional_story_ids(self, new_additional_story_ids):
+                uncategorized_skill_ids list.
+        """
+        if uncategorized_skill_id not in self.uncategorized_skill_ids:
+            raise Exception(
+                'The skill id %s is not present in the topic.'
+                % uncategorized_skill_id)
+        self.uncategorized_skill_ids.remove(uncategorized_skill_id)
+
+    def get_subtopic_index(self, subtopic_id):
+        """Gets the index of the subtopic with the given id in the subtopics
+        list.
+
+        Args:
+            subtopic_id: str. The id of the subtopic for which the index is to
+                be found.
+
+        Returns:
+            int or None. Returns the index of the subtopic if it exists or else
+                None.
+        """
+        for ind, subtopic in enumerate(self.subtopics):
+            if subtopic.id == subtopic_id:
+                return ind
+        return None
+
+    def add_subtopic(self, subtopic_id, title):
+        """Adds a subtopic with the given id and title.
+
+        Args:
+            subtopic_id: str. The id for the new subtopic.
+            title: str. The title for the new subtopic.
+
+        Raises:
+            Exception. A subtopic with the given id already exists.
+        """
+        if self.get_subtopic_index(subtopic_id) is not None:
+            raise Exception(
+                'A subtopic with id %s already exists. ' % subtopic_id)
+        self.subtopics.append(
+            Subtopic.create_default_subtopic(subtopic_id, title))
+
+    def delete_subtopic(self, subtopic_id):
+        """Deletes the subtopic with the given id and adds all its skills to
+        uncategorized skill ids section.
+
+        Args:
+            subtopic_id: str. The id of the subtopic to remove.
+
+        Raises:
+            Exception. A subtopic with the given id doesn't exists.
+        """
+        subtopic_index = self.get_subtopic_index(subtopic_id)
+        if subtopic_index is None:
+            raise Exception(
+                'A subtopic with id %s doesn\'t exist. ' % subtopic_id)
+        for skill_id in self.subtopics[subtopic_index].skill_ids:
+            self.uncategorized_skill_ids.append(skill_id)
+        del self.subtopics[subtopic_index]
+
+    def update_subtopic_title(self, subtopic_id, new_title):
+        """Updates the title of the new subtopic.
+
+        Args:
+            subtopic_id: str. The id of the subtopic to edit.
+            new_title: str. The new title for the subtopic.
+
+        Raises:
+            Exception. The subtopic with the given id doesn't exist.
         """
-        self.skill_ids = new_skill_ids
+        subtopic_index = self.get_subtopic_index(subtopic_id)
+        if subtopic_index is None:
+            raise Exception(
+                'The subtopic with id %s does not exist.' % subtopic_id)
+        self.subtopics[subtopic_index].title = new_title
+
+    def move_skill_id_to_subtopic(self, subtopic_id, new_skill_id):
```

Comment:
Sorry, my bad -- per comment on commit log entry I think we should pass in the old_subtopic_id as well.

Note that old/new subtopic id could be None (which implies "uncategorized"). This should be explained in the docstring. (Alternatively you could have a separate method for remove_skill_from_subtopic that moves a skill to the uncategorized list -- that may be better/clearer, actually. So only the old_subtopic_id has the option of being None.)

Also new_skill_id should be skill_id, right? The skill already exists.

Reply:
Yeah, done. I renamed command to CMD_MOVE_SKILL_ID. So, is that fine, or should a remove_skill_from_subtopic() command be there also?

Reply:
I was thinking about this just now, too! I think CMD_REMOVE_SKILL_FROM_SUBTOPIC makes sense.

Reply:
Ok, done.